modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/TransactionStateResolver.java

```diff
@@ -138,6 +141,26 @@ public CompletableFuture<TransactionMeta> resolveTxState(
             UUID txId,
             ReplicationGroupId commitGrpId,
             @Nullable HybridTimestamp timestamp
+    ) {
+        return resolveTxState(txId, commitGrpId, timestamp, AWAIT_PRIMARY_REPLICA_TIMEOUT, SECONDS);
+    }
+
+    /**
+     * Resolves transaction state locally, if possible, or distributively, if needed.
+     *
+     * @param txId Transaction id.
+     * @param commitGrpId Commit partition group id.
+     * @param timestamp Timestamp.
+     * @param awaitPrimaryReplicaTimeout Timeout for awaiting primary replica.
+     * @param awaitPrimaryReplicaTimeUnit Time unit for awaiting primary replica timeout.
+     * @return Future with the transaction state meta as a result.
+     */
+    public CompletableFuture<TransactionMeta> resolveTxState(
+            UUID txId,
+            ReplicationGroupId commitGrpId,
+            @Nullable HybridTimestamp timestamp,
+            long awaitPrimaryReplicaTimeout,
```

Comment:
Within the context of given method, I'd rather name the param as "awaitCommitPartitionAvailabilityTimeout" or similar.